index.js

```diff
@@ -269,7 +269,7 @@ ExtractTextPlugin.prototype.apply = function(compiler) {
 			async.forEach(chunks, function(chunk, callback) {
 				var extractedChunk = extractedChunks[chunks.indexOf(chunk)];
 				var shouldExtract = !!(options.allChunks || isInitialOrHasNoParents(chunk));
-				async.forEach(chunk.modules.slice(), function(module, callback) {
+				async.forEach(chunk.mapModules(function(c) { return c; }), function(module, callback) {
```

Comment:
`mapModules` map functions on each modules so the iteratee will be called with each module. See https://github.com/webpack/webpack/pull/4764/files#diff-8941681d920ac4feda44f522b986c2d0R115